tests/dbos.test.ts

```diff
@@ -275,8 +275,114 @@ describe('dbos-tests', () => {
       );
     });
   });
+
+  test('workflow-withWorkflowTimeout', async () => {
+    const workflowID: string = randomUUID();
+    await DBOS.withNextWorkflowID(workflowID, async () => {
+      await DBOS.withWorkflowTimeout(100, async () => {
+        await expect(DBOSTimeoutTestClass.blockedWorkflow()).rejects.toThrow(
+          new DBOSWorkflowCancelledError(workflowID),
+        );
+      });
+    });
+    const status = await DBOS.getWorkflowStatus(workflowID);
+    expect(status?.status).toBe(StatusString.CANCELLED);
+  });
+
+  test('workflow-timeout-startWorkflow-params', async () => {
+    const workflowID = randomUUID();
+    const handle = await DBOS.startWorkflow(DBOSTimeoutTestClass, { workflowID, timeout: 100 }).blockedWorkflow();
+    await expect(handle.getResult()).rejects.toThrow(new DBOSWorkflowCancelledError(workflowID));
+    const status = await DBOS.getWorkflowStatus(workflowID);
+    expect(status?.status).toBe(StatusString.CANCELLED);
+  });
+
+  test('parent-workflow-withWorkflowTimeout', async () => {
+    const workflowID: string = randomUUID();
+    await DBOS.withNextWorkflowID(workflowID, async () => {
+      await DBOS.withWorkflowTimeout(100, async () => {
+        await expect(DBOSTimeoutTestClass.blockingParentStartWF()).rejects.toThrow(
+          new DBOSWorkflowCancelledError(workflowID),
+        );
+      });
+    });
+    const status = await DBOS.getWorkflowStatus(workflowID);
+    expect(status?.status).toBe(StatusString.CANCELLED);
+  });
+
+  test('parent-workflow-timeout-startWorkflow-params', async () => {
+    const workflowID = randomUUID();
+    const handle = await DBOS.startWorkflow(DBOSTimeoutTestClass, { workflowID, timeout: 100 }).blockingParentStartWF();
+    await expect(handle.getResult()).rejects.toThrow(new DBOSWorkflowCancelledError(workflowID));
+    const status = await DBOS.getWorkflowStatus(workflowID);
```

Comment:
Similar to the client test. Let's also check that the child workflows are properly cancelled, making sure the deadline is propagated to all children.